drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationDocksHandler.java

```diff
@@ -36,13 +37,19 @@
         extends AbstractWorkbenchDocksHandler {
 
     @Inject
-  private   AuthoringWorkbenchDocks authoringWorkbenchDocks;
+    private AuthoringWorkbenchDocks authoringWorkbenchDocks;
+
+    @Inject
+    private PlaceManager placeManager;
 
     private UberfireDock toolsDock;
     private UberfireDock reportDock;
 
     @Override
     public Collection<UberfireDock> provideDocks(final String perspectiveIdentifier) {
+
+        placeManager.closePlace("org.kie.guvnor.TestResults");
```

Comment:
@Rikkola 
Why is this needed ? I have the feeling that injecting and using PlaceManager here somehow hides something that should be managed elsewhere - just an impression. Beside that, even if technically it is possible to inject stuff (and  all that works only because they are sort of "singleton") this creates a sort a complex coupling that we should strive to avoid, IMO.

Reply:
This is since there might be a TestResult screen open. For example when running all the test the results are opened into a panel. Then when looking at the editor it tries to open the same screen in two locations. One in regular panel and another in docks.

To get rid of this I guess I would have to use similar approach to what you had where the editor and the dock instance are tied together.

Reply:
Actually does not make sense to call this here. Not sure why it was here. Removing and testing.